### ISSUE TITLE
[Bug]: Fluent UI split buttons violate the nested-interactive rule

### ISSUE THREAD

**Issue by christopher-o-toole** [microsoft/fluentui#25541]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Environment: Microsoft Edge version 107.0.1418.35

====

This accessibility issue was found using Accessibility Insights for Web 2.35.0 (axe-core 4.4.1), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/button

## Actual Behavior
![image](https://user-images.githubusercontent.com/22433652/200071716-6aeeeaa0-b7af-49c6-8e74-778c53cb0614.png)

The screenshot above shows three of the four nested-interactive violations on the page. The issues from left to right are as follows:
- Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (.css-341[aria-roledescription="split button"][role="button"]:nth-child(1))
- Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (.css-341[aria-roledescription="split button"][role="button"]:nth-child(2))
- Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (.css-341[aria-roledescription="split button"][role="button"]:nth-child(3))

## Expected Behavior
Split button components should adhere to the Web Content Accessibility Guidelines (WCAG).

### Requested priority

High

### Products/sites affected

Azure Notebooks Component

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
Hi! This is actually by design, the issue is that we have a control that is a single keyboard tab/arrow stop, but has two actions (the primary button and opening a menu). We can't remove either the outer interactive role or the inner ones without degrading practical accessibility, and we can't switch to a two-button and two-tab-stop approach without introducing a breaking change.

We've done work to mitigate the practical accessibility issues with this approach, so it while it may be quirky with various screen readers, it shouldn't violate WCAG or be blocking to anyone. More context here: https://github.com/microsoft/fluentui/issues/18484#issuecomment-861723118

The v9 splitbutton does use two separate buttons with separate tab/arrow stops 😊
